### PR TITLE
Send email on commit

### PR DESCRIPTION
**Pull request by adibo**
I propose to use `on_commit` for sending out activation emails.

This could be useful when e.g. sending out activation emails are delegated to a celery task. Potential race condition exists, when a user object has yet not been committed to a db while a task broker already attempts to execute email sending. 
A similar problem is described here: https://dev.to/k4ml/django-fixing-race-condition-when-queuing-with-oncommit-hook-7ae

Since `django-registration` relies on Django 1.11+, we could safely use `on_commit` to avoid such situations, as sugested by the post.

The only caveat is that (some) tests would need to use `TransactionTestCase`. 
All the necessary changes are proposed in this PR.

### CODE REVIEW

File: test_app/settings_test.py [hunk 1]
```diff
@@ -60,3 +60,5 @@
     ('admin1', 'registration_admin1@mail.server.com'),
     ('admin2', 'registration_admin2@mail.server.com'),
 )
+
+ACCOUNT_ACTIVATION_DAYS = 7
```
Comment (joshblum) [inline test_app/settings_test.py]: this shouldn't be necessary since we use `override_settings` on the test cases. why is this change necessary? are the overrides no longer working?

**Comment by joshblum** [comment]
thanks for the PR! this change looks good, will accept once the testing issue is resolved

File: test_app/settings_test.py (hunk 1 above)
Comment (adibo) [reply test_app/settings_test.py]: The tests did not pass on a fresh fork from master, before I made any changes (fresh clone). 
Adding the `ACCOUNT_ACTIVATION_DAYS` to `settings_test.py` helped. 
I did not investigate it further, though. I assumed that it was  accidentally removed.
Let's look into it!
Could you check the results of `invoke test` on master?

Comment (joshblum) [reply test_app/settings_test.py]: seems to be a weird interaction between pytest/pytest-django and our setup in recent versions.

Comment (adibo) [reply test_app/settings_test.py]: If I run tests without `ACCOUNT_ACTIVATION_DAYS` in the `settings_test.py` 5 tests from `default_backend.DefaultBackendViewTests` fail (only from that file). 
If I then apply `@override_settings(ACCOUNTS_ACTIVATION_DAYS=7)` to those (failing) tests, they pass.
This result suggests that there is some problem with `@override_setting` when used jointly at class and method/test level.

Without going any deeper, one way to resolve the issue is to add the missing line to those tests. We then can safely remove `ACCOUNT_ACTIVATION_DAYS` from the `settings_test.py`.
Any preference or suggestion?

Comment (joshblum) [reply test_app/settings_test.py]: i'm ok with leaving the setting in `settings_test.py`, thanks for the patch!

Comment (adibo) [reply test_app/settings_test.py]: Great! Thanks!
